esp/esdllib/esdl_script.cpp

```diff
@@ -234,6 +239,414 @@ interface IEsdlTransformOperationHttpHeader : public IInterface
+        m_resultsetTag.setown(compileOptionalXpath(stag.getValue("resultset-tag")));
+
+        m_server.setown(compileOptionalXpath(stag.getValue("server")));
+        m_user.setown(compileOptionalXpath(stag.getValue("user")));
+        m_password.setown(compileOptionalXpath(stag.getValue("password")));
+        m_database.setown(compileOptionalXpath(stag.getValue("database")));
+
+        //script can set any MYSQL options using an attribute with the same name as the option enum, for example
+        //    MYSQL_SET_CHARSET_NAME="'latin1'" or MYSQL_SET_CHARSET_NAME="$charset"
+        //
+        int attCount = stag.getLength();
+        for (int i=0; i<attCount; i++)
+        {
+            const char *attName = stag.getLocalName(i);
+            if (attName && hasPrefix(attName, "MYSQL_", false))
+            {
+                Owned<ICompiledXpath> attXpath = compileOptionalXpath(stag.getValue(i));
+                if (attXpath)
+                {
+                    m_mysqlOptionNames.append(attName);
+                    m_mysqlOptionXpaths.append(*attXpath.getClear());
+                }
+            }
+        }
+
+        int type = 0;
+        while((type = xpp.next()) != XmlPullParser::END_DOCUMENT)
+        {
+            switch(type)
+            {
+                case XmlPullParser::START_TAG:
+                {
+                    StartTag stag;
+                    xpp.readStartTag(stag);
+                    const char *op = stag.getLocalName();
+                    if (isEmptyString(op))
+                        esdlOperationError(ESDL_SCRIPT_Error, m_tagname, "unknown error", m_traceName, !m_ignoreCodingErrors);
+                    if (streq(op, "bind"))
+                        m_parameters.append(*new CEsdlTransformOperationMySqlBindParmeter(xpp, stag, prefix));
+                    else if (streq(op, "sql"))
+                        readFullContent(xpp, m_sql);
+                    else
+                        xpp.skipSubTreeEx();
+                    break;
+                }
+                case XmlPullParser::END_TAG:
+                case XmlPullParser::END_DOCUMENT:
+                    return;
+            }
+        }
+
+        if (!m_section)
+            m_section.setown(compileXpath("'temporaries'"));
+        StringBuffer errmsg;
+        if (m_name.isEmpty())
+            buildMissingMySqlParameterMessage(errmsg, "name");
+        if (!m_server)
+            buildMissingMySqlParameterMessage(errmsg, "server");
+        if (!m_user)
+            buildMissingMySqlParameterMessage(errmsg, "user");
+        if (!m_database)
+            buildMissingMySqlParameterMessage(errmsg, "database");
+        if (m_sql.isEmpty())
+            buildMissingMySqlParameterMessage(errmsg, "sql");
+        if (errmsg.length())
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname, errmsg, m_traceName, !m_ignoreCodingErrors);
+    }
+    void getSecretValues()
+    {
```

Comment:
As per our conversation, this is unused.

Reply:
removed.